homework/vector-of-shared-ptrs/vectorFunctions.cpp

```diff
@@ -0,0 +1,45 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+#include "vectorFunctions.hpp"
+
+std::vector <std::shared_ptr<int>> generate(int count) {
+
+    std::vector <std::shared_ptr<int>> vec;
+    for (int i = 0; i < count; ++i) {        
+        vec.push_back(std::make_shared<int>(i));
+    }
+    return vec;
+}
+
+void print(std::vector <std::shared_ptr<int>> print_vec) {
+
+    for (auto el : print_vec) {
+        if (el != nullptr){
+            std::cout << *el <<  " ";
+        }
```

Comment:
if (value) is enough. This will call implicit `bool` operator which compare this ptr with `nullptr`